liquidctl/driver/lianli_uni.py

```diff
@@ -0,0 +1,239 @@
+            self.pwm_channels[channel] = desired_state  # Update state
+        except Exception as e:
+            _LOGGER.warning("Error writing to device: %s", e)
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set a fixed speed for the specified channel.
+
+        Parameters:
+            channel: str or int - The name of the channel (e.g., 'channel1') or the zero-based index of the channel
+            duty: int or float - The desired speed percentage (0-100)
+        """
+        if isinstance(channel, str):
+            channel_index = self._extract_channel_index(channel)
+        else:
+            channel_index = channel
+
+        if not _MIN_CHANNEL <= channel_index <= _MAX_CHANNEL:
+            raise ValueError(
+                f"channel must be between {_MIN_CHANNEL} and {_MAX_CHANNEL} (zero-based index)"
+            )
+
+        # Check if PWM is on for the channel
+        if self.pwm_channels[channel_index]:
+            _LOGGER.warning(
+                f"Cannot set fixed speed for Channel {channel_index + 1}: PWM is enabled. "
+                "Please disable PWM first to set a fixed speed."
+            )
+            return
+
+        duty = clamp(duty, 0, 100)
+        speed_byte = self._calculate_speed_byte(duty)
+        command = [224, channel_index + 32, 0, speed_byte]
+        _LOGGER.debug(
+            "Setting fixed speed for channel %d: duty %.1f%%, command %s",
+            channel_index,
+            duty,
+            command,
+        )
+        try:
+            self.device.write(command)
+        except Exception as e:
+            _LOGGER.error("Error writing to device: %s", e)
+        time.sleep(0.1)  # Delay to prevent race conditions
+
+        self.channel_speeds[channel_index] = duty
+
+        _LOGGER.info(f"Fan speed for Channel {channel_index + 1} set to {duty}%")
+
+    def _calculate_speed_byte(self, speed):
+        """Calculate the speed byte based on the device type and desired speed.
+
+        Parameters:
+            speed: int or float - The desired speed percentage (0-100)
+
+        Returns:
+            int - The calculated speed byte to send to the device
+        """
+        if self.device_type in ["SL", "AL"]:
+            if speed == 0:
+                return 40
+            else:
+                speed_byte = int((800 + (11 * speed)) / 19) & 0xFF
+        elif self.device_type == "SLI":
+            if speed == 0:
+                return 10
+            else:
+                speed_byte = int((250 + (17.5 * speed)) / 20) & 0xFF
+        elif self.device_type in ["SLV2", "ALV2"]:
+            if speed == 0:
+                return 7
+            else:
+                speed_byte = int((200 + (19 * speed)) / 21) & 0xFF
+        else:
+            raise NotSupportedByDevice(
+                f"Unsupported device type: {self.device_type}"
+            )
+        return speed_byte
+
+    def _extract_channel_index(self, channel):
```

Comment:
In case this is kept, it could be moved to `util`. Just mark it as unstable (like some other functions in that module).